neqo-transport/src/qlog.rs

```diff
@@ -241,66 +265,63 @@ pub enum QlogMetric {
 pub fn metrics_updated(qlog: &mut NeqoQlog, updated_metrics: &[QlogMetric]) {
     debug_assert!(!updated_metrics.is_empty());
 
-    qlog.add_event(|| {
-        let mut min_rtt: Option<u64> = None;
-        let mut smoothed_rtt: Option<u64> = None;
-        let mut latest_rtt: Option<u64> = None;
-        let mut rtt_variance: Option<u64> = None;
-        let mut max_ack_delay: Option<u64> = None;
-        let mut pto_count: Option<u64> = None;
+    qlog.add_event_data(|| {
+        let mut min_rtt: Option<f32> = None;
+        let mut smoothed_rtt: Option<f32> = None;
+        let mut latest_rtt: Option<f32> = None;
+        let mut rtt_variance: Option<f32> = None;
+        let mut pto_count: Option<u16> = None;
         let mut congestion_window: Option<u64> = None;
         let mut bytes_in_flight: Option<u64> = None;
         let mut ssthresh: Option<u64> = None;
         let mut packets_in_flight: Option<u64> = None;
-        let mut in_recovery: Option<bool> = None;
         let mut pacing_rate: Option<u64> = None;
 
         for metric in updated_metrics {
             match metric {
-                QlogMetric::MinRtt(v) => min_rtt = Some(u64::try_from(v.as_millis()).unwrap()),
+                QlogMetric::MinRtt(v) => min_rtt = Some(v.as_secs_f32() * 1000.0),
```

Comment:
Feature request for you: can you take these as `Duration` rather than `f32`?

Reply:
The qlog spec data model defines the value as float32; see https://quicwg.org/qlog/draft-ietf-quic-qlog-quic-events.html#section-7.2

The qlog crate data model tries to be as close to the spec as possible. I don't see much value in e.g. turning the crate to support enum foo(spec_def(f32), my_own_thing(Duration) because it would have to be converted to f32 on way or another.

Reply:
Yeah, I get that, but Rust native time types are more ergonomic.

You could maybe double-dip by implementing a trait:

```rust
trait IntoQlogTime {
  fn into_qlog_time(self) -> f32;
}
impl IntoQlogTime for f32 { ... }
impl IntoQlogTime for Duration { ... }
```

Though that helps more with if you also provide functions for instantiating the objects rather than making the fields public as you do (because then you can do `x: impl IntoQlogTime` arguments).

Reply:
Ta, I'm happy to add some syntactic sugar as a feature request